### PR TITLE
#1300 (Bug: Array to string conversion while deleting extras)

### PR DESCRIPTION
**Pull request by vytenizs**

### CODE REVIEW

File: src/Backend/Core/Engine/Model.php [hunk 1]
```diff
@@ -198,18 +198,24 @@ public static function deleteExtra($module = null, $type = null, array $data = n
...
+            $delete = true;
+
             // match by parameters
             if ($data !== null && $extra['data'] !== null) {
                 $extraData = (array) unserialize($extra['data']);
 
-                // skip extra if parameters do not match
-                if (count(array_intersect($data, $extraData)) !== count($data)) {
-                    continue;
+                // delete extra if parameters do not match
```
Comment (WouterSioen) [inline src/Backend/Core/Engine/Model.php]: I'd change the comment here to "**do not** delete extra if the parameters do not match"

File: src/Backend/Core/Engine/Model.php [hunk 2]
```diff
@@ -198,18 +198,24 @@ public static function deleteExtra($module = null, $type = null, array $data = n
 
         // loop found extras
         foreach ($extras as $extra) {
+            $delete = true;
```
Comment (WouterSioen) [inline src/Backend/Core/Engine/Model.php]: I'd rename this variable to "deleteExtra", this is more explicit

**Comment by WouterSioen** [comment]
can you still squash the commits? (this is explained in the CONTRIBUTING.md guide)

File: src/Backend/Core/Engine/Model.php [hunk 3]
```diff
@@ -198,18 +198,24 @@ public static function deleteExtra($module = null, $type = null, array $data = n
...
+
             // match by parameters
             if ($data !== null && $extra['data'] !== null) {
                 $extraData = (array) unserialize($extra['data']);
 
-                // skip extra if parameters do not match
-                if (count(array_intersect($data, $extraData)) !== count($data)) {
-                    continue;
+                // do not delete extra if parameters do not match
+                foreach ($data as $dataKey => $dataValue) {
```
Comment (WouterSioen) [inline src/Backend/Core/Engine/Model.php]: can't this be done more efficiently using http://php.net/manual/en/function.array-intersect-assoc.php?

Comment (WouterSioen) [inline src/Backend/Core/Engine/Model.php]: This is possible using http://php.net/manual/en/function.array-diff-assoc.php. This is probably more efficient

Comment (vytenizs) [inline src/Backend/Core/Engine/Model.php]: Have you tried this with multidimensional arrays?

Comment (vytenizs) [inline src/Backend/Core/Engine/Model.php]: Ive read that all array function acts tricky with multidimensional arrays

**Pull request merged**